### PR TITLE
Querier ignore 'The specified key does not exist' error

### PR DESCRIPTION
**Pull request by hczhu-db**
# Testing
## Deployed to a test region

### CODE REVIEW

File: pkg/store/proxy.go [hunk 1]
```diff
@@ -421,11 +426,17 @@ func (s *ProxyStore) Series(originalRequest *storepb.SeriesRequest, srv storepb.
...
 			totalFailedStores++
 			maxWarningBytes := 2000
+			shouldIgnore := strings.Contains(resp.GetWarning(), "The specified key does not exist")
 			warning := resp.GetWarning()[:min(maxWarningBytes, len(resp.GetWarning()))]
-			level.Error(s.logger).Log("msg", "Store failure with warning", "warning", warning)
+			level.Error(s.logger).Log("msg", "Store failure with warning", "warning", warning, "should_ignore", shouldIgnore)
 			// Don't have group/replica keys here, so we can't attribute the warning to a specific store.
 			s.metrics.storeFailureCount.WithLabelValues("", "").Inc()
-			if r.PartialResponseStrategy == storepb.PartialResponseStrategy_GROUP_REPLICA {
+			if shouldIgnore {
```
Comment (jnyi) [inline pkg/store/proxy.go]: should this be within the if condition here: 
```
if r.PartialResponseStrategy == storepb.PartialResponseStrategy_GROUP_REPLICA 
```

Comment (hczhu-db) [reply pkg/store/proxy.go]: We can do that, though all our deployments are on group_replica strategy. I'll update.

File: pkg/store/proxy.go [hunk 2]
```diff
@@ -419,24 +424,31 @@ func (s *ProxyStore) Series(originalRequest *storepb.SeriesRequest, srv storepb.
 		resp := respHeap.At()
 
 		if resp.GetWarning() != "" {
-			totalFailedStores++
```
Comment (jnyi) [inline pkg/store/proxy.go:422]: any reason to remove this?

**Review by jnyi — APPROVED** [review]
lgtm

**Pull request merged**